APP.md

```diff
@@ -21,6 +21,30 @@ api.fetch('/users', function(err, result) {
     console.log(result)
 })
 ```
+The user should also be able to define preexsiting urls and have a method that corresponds to the request. These methods should use promises as well. For example: 
```

Comment:
Would you make custom methods just for an "user" endpoint? I would like your other solution, the one below, much more.

Reply:
The idea here is for endpoints that correspond to one thing. For a thing like "users", a resource would make much more sense. But for things where you can only do one action, like getting a user's friend list on facebook you would only need 
```javascript
api.get("user_friendlist", "...");
```
Rather than making a resource because that isn't necessary. So you would need both resources and single methods.

Reply:
I just realised that POST and PUT requests wouldn't make much sense in the current state.

POST and PUT requests are often used for creating and updating data from a server.
GET is used to request data from a server.

You can currently get cached GET data, which makes sense, because it's simply data from the past.
But why would you want to cache PUT and POST data? These require immediate action from the server.

Reply:
I agree. I guess I was thinking that this would evolve into a more of an API client than just a simple caching tool. Plus you can cache when a request is idempotent, like for PUT and DELETE.